modules/storage/filesystem/device.go

```diff
@@ -0,0 +1,186 @@
+package filesystem
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	log "github.com/rs/zerolog/log"
+)
+
+type DeviceManager interface {
+	Device(ctx context.Context, device string) (Device, error)
```

Comment:
Please document how each method of the interface should work and what's its purpose